src/modm/platform/clock/stm32/clock.cpp.in

```diff
@@ -261,7 +263,6 @@ modm::platform::ClockControl::enablePll(PllSource source,
 	if (uint32_t(pllPrediv - 1) & 0x1) tmp |= RCC_CFGR_PLLXTPRE;
 #endif
```

Comment:
The line above still modifies `tmp`!

Reply:
To my understanding this line is never used because the else branch of `#ifdef {{mask}}` is never taken. This `ifdef` is inside of `%%if prediv` which is true for all F0, F3 and F10{0,5,7}. All these controllers define `{{mask}}` in their headers (checked with `grep`).

`RCC_CFGR_PLLXTPRE` (bit 17 in `RCC->CFGR`) refers to the same hardware bit as bit 0 in `RCC->CFGR2` (if available). Writing to either bit will also write the other. This is the reason why writing to `RCC->CFGR` changes `RCC->CFGR2`.

According to the F1 reference manual the fix should also work for F1:

![image](https://user-images.githubusercontent.com/25187160/46630454-82f35580-cb44-11e8-8b57-1e53e4a406b1.png)

The line `if (uint32_t(pllPrediv - 1) & 0x1) tmp |= RCC_CFGR_PLLXTPRE;` would only make sense for controllers without `prediv` and `RCC->CFGR2` (F10{1,2,3}) to configure the x1/x2 prescaler.

Reply:
Ahh! Thanks for stepping in @chris-durand. This is quite subtle ... I'm still not quite sure how to fix this. Should we handle the case of F10{1,2,3} separately and unify the other cases to just setting `RCC->CFGR`?

Reply:
> Should we handle the case of F10{1,2,3} separately and unify the other cases to just setting RCC->CFGR?

Your initial fix is correct. The upper 3 bits of the prescaler (in controllers with a flexible 1..16 prescaler) have to be set using `RCC->CFGR2`. It is just the LSB that is aliased by `RCC_CFGR_PLLXTPRE` in `RCC->CFGR`. Thus, when you set `CFGR2` after `CFGR` all should be fine. 

If I understand this correctly the line @salkinium mentioned is never used but won't do any harm.

F10{1,2,3} should also work right now, the /2 prescaler is simply not accessible and fixed to 1.
To add it we would need an additional function parameter [here](https://github.com/modm-io/modm/blob/4f1d57b3c10fcecd90cbccc6353270bc83444536/src/modm/platform/clock/stm32/clock.cpp.in#L235). Unfortunately, it would break all code that specifies a custom value for `waitCycles`. I'm not sure if anyone actually does that. Putting the prescaler after `waitCycles` would be an equally ugly option.

Reply:
> Unfortunately, it would break all code that specifies a custom value for waitCycles. 

That's not a reason! You get what you pay!1!! Bwaharhar.

> I'm not sure if anyone actually does that.

Nobody uses that, since everybody copies our examples/board files and they don't use that. Not sure what a useful error handling strategy would be for that anyways, if your clock is screwed that's pretty bad.